corehq/apps/registration/tests/test_utils.py

```diff
@@ -94,17 +105,26 @@ def test_domain_is_not_active_for_new_user(self):
         self.assertFalse(domain.is_active)
 
     @suspend(domain_tombstone_patch)
-    @mock.patch('corehq.apps.registration.utils._setup_subscription', _issue_initializing_domain)
+    @mock.patch.object(BillingContactInfo.objects, 'get_or_create', _issue_initializing_domain)
     def test_subscription_exception_raises_error_and_domain_is_deleted(self):
         # We want to ensure that errors during the Subscription initialization process
         # do not result in incomplete domains (a domain without a Subscription)
+        def check_preconditions():
+            # verify that accounting models created during domain setup do not exist
+            Subscription.clear_caches(domain_name)
+            self.assertFalse(domain_or_deleted_domain_exists(domain_name))
+            self.assertIsNone(BillingAccount.get_account_by_domain(domain_name))
+            self.assertIsNone(Subscription.get_active_subscription_by_domain(domain_name))
+
         domain_name = 'subscription-failed'
         self.addCleanup(_cleanup_domain, domain_name)
+        check_preconditions()
```

Comment:
nit: it doesn't hurt so fine to keep it, but this usually reflects uncertainty in test cleanup right? And now that test cleanup is resolved, this shouldn't be necessary?

(I also think I did this in one of my recent tests 😅)

Reply:
Correct, although it would be very confusing if some other test did not clean up properly, causing the final `check_preconditions()` to fail at the end of this test although not due to anything wrong with the code this is testing. Having this here should clearly highlight that situation.